src/simple_server/simple_game.clj

```diff
@@ -1,23 +1,53 @@
 (ns simple-server.simple-game)
 
 
-(def game-in-progress (atom nil))
+(def game-in-progress (atom {}))
 
-(defn new-game! []
+(defn reset-game
+  "Reset the game status for a user."
+  [game-data username]
+  (swap! game-data assoc username {:value (+ 1 (rand-int 10)) :times 0}))
+
+(defn get-value
+  "Return the value for this round for a user."
+  [game-data username]
+  (get-in @game-data [username :value]))
+
+(defn get-times
+  "Return the number of guessing that a user has done."
+  [game-data username]
+  (get-in @game-data [username :times]))
+
+(defn inc-times
+  [game-data username]
+  (swap! game-data update-in [username :times] #(inc %)))
+
+(defn new-game! [name]
   ;; Make our new game:
-  (reset! game-in-progress (+ 1 (rand-int 10)))
+  (reset-game game-in-progress name)
   :ok)
 
-(defn guess-answer [guess]
-  (cond
-    (nil? guess) nil
+(defn guess-answer
+  [name guess]
+  (let [value (get-value game-in-progress name) times (get-times game-in-progress name)]
+    (cond
+      (or (nil? guess) (not= java.lang.Long (type guess))) nil
+      ;;you have to start a new game before guessing.
+      (nil? value) :game-not-started
+
+      (or (> times 4) (and (= times 4) (not= guess value)))
+      (do (reset-game game-in-progress name)
+          :game-over)
+
+      (= guess value)
+      (do (reset-game game-in-progress name)
+          :win)
 
-    (= guess @game-in-progress)
-    (and (reset! game-in-progress (+ 1 (rand-int 10)))
-         :game-over)
+      (< guess value)
+      (do (inc-times game-in-progress name)
+          :too-low)
 
-    (< guess @game-in-progress)
-    :too-low
+      (> guess value)
+      (do (inc-times game-in-progress name)
+          :too-high))))
```

Comment:
All in all, a great effort; your best one yet, I'd say.

I think this shows your previous experience with web programming; you clearly already have a good mental model of how all this stuff should fit together.  Great stuff!

Reply:
haha, thanks. I just try to apply what I have learnt in the ring and complojure module.